### PR TITLE
Custom error for config validation failure.

### LINKED ISSUE #92: Re-raise errors during config file validation as ValidationError
1. create errors.py at same level at _init_.py
2. create a class ValidationError
3. raise ValidationError from config_parser.py where there is empty return and pass the appropriate error message eg. raise ValidationError('Config file has errors - {errors}')
4. perform step 3 when the file is invalid as well

### PR DESCRIPTION
**Pull request by Manojkumar-Chandru-ML**
Created ValidationError class inside errors.py to reraise exceptions occurring during validation of config.yaml.
Resolves #92

### CODE REVIEW

File: simulator/src/config_parser.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import yaml
 from cerberus import Validator
 import constants as const
-
+from errors import ValidationError
 
 class Config:
```
Comment (Siddharth-Rai-ML) [inline simulator/src/config_parser.py:9]: there should always be 2 empty lines before class definition

File: simulator/src/config_parser.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def parse_config(config_file_path):
         fp.close()
         sys.stdout.write(e)
         sys.stdout.write("Could not read data, please check if all fields are filled")
-        return
+        raise ValidationError(str(e))
```
Comment (Siddharth-Rai-ML) [inline simulator/src/config_parser.py:66]: sys.stdout.write(e) can be removed, all the messages can be combined and raised as ValidationError.

eg. ValidationError('error reading config file - ' + str(e))

File: simulator/src/errors.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+class ValidationError(Exception):
+    pass
```
Comment (Siddharth-Rai-ML) [inline simulator/src/errors.py:2]: Add an empty line the end of file

File: simulator/src/config_parser.py [hunk 4]
```diff
@@ -84,7 +84,7 @@ def parse_config(config_file_path):
     else:
         sys.stdout.write(str(errors))
         sys.stdout.write("Please pass a Valid config file")
-        exit()
+        raise ValidationError(str(e))
```
Comment (Siddharth-Rai-ML) [inline simulator/src/config_parser.py:87]: There is no variable called e

Here it should be:

ValidationError('Error validating config file - ' + str(errors))

sys.stdout can be removed

File: simulator/src/errors.py (hunk 3 above)
Comment (Siddharth-Rai-ML) [inline simulator/src/errors.py:2]: Add docstring:

"""Raised when errors are encountered in cofig file validation"""

**Review by Siddharth-Rai-ML — CHANGES_REQUESTED** [review]
Please have a look

File: simulator/src/errors.py (hunk 3 above)
Comment (Manojkumar-Chandru-ML) [reply simulator/src/errors.py]: I have added extra line

**Comment by Akhil-Nair-ML** [comment]
LGTM

**Pull request merged**